apps/cowswap-frontend/src/modules/usdAmount/utils/usdcPriceLoader.ts

```diff
@@ -0,0 +1,20 @@
+import { USDC } from '@cowprotocol/common-const'
+import { FractionUtils } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Fraction } from '@uniswap/sdk-core'
+
+import { getCowProtocolNativePrice } from '../apis/getCowProtocolNativePrice'
+
+export function usdcPriceLoader(chainId: SupportedChainId): () => Promise<Fraction | null> {
+  let usdcPricePromise: Promise<number | null> | null = null
+
+  return async () => {
+    // Cache the result to avoid fetching it multiple times
+    if (!usdcPricePromise) {
+      usdcPricePromise = getCowProtocolNativePrice(USDC[chainId])
+    }
+
+    const usdcPrice = await usdcPricePromise
+    return typeof usdcPrice === 'number' ? FractionUtils.fromNumber(usdcPrice) : null
```

Comment:
what other thing would it be if not a number?

Reply:
Probably a check to make TS happy.
The fn was just moved, code was already there.